src/components/ui/checkbox-group.svelte

```diff
@@ -12,14 +12,18 @@
   export let items = [];
   export let value = [];
   export let uniqueKey = null;
+  export let max = null;
 
   $: parsedItems = parseItems(items);
   $: selected = parseValues(value, parsedItems, {uniqueKey, multiple: true});
 
   let dispatch = createEventDispatcher();
-  const changeCheckbox = item => {
+  const changeCheckbox = (item, e) => {
     if (selected.some(x => x.id === item.id)) selected = selected.filter(x => x.id !== item.id);
-    else selected.push(item);
+    else {
+      if (!max || selected.length < max) selected.push(item);
+      else e.target.checked = false;
+    }
```

Comment:
I think it would be better in terms of UX to actually disable the unchecked checkboxes themselves when 3 are checked.
Something along the lines of `disabled={!selected.some(x => x.id === item.id) && max != null && selected.length >= max}` on the input element (and maybe refactor the first clause into a function and the other two into a reactive variable? that line is a bit long)

Reply:
Yeah, maybe. But not now)